infor-design/enterprise-wc
# 1065 - Separate IdsDropdown's Popup into its own component / integrate with IdsDataGridFilters

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR makes some changes to IdsDropdown to allow it to break out of IdsDataGrid's overflow containment while being used as a filter:
- Separates the IdsDropdown's popup into its own new component, IdsDropdownList.  This new component is the new home to IdsListBox/IdsListBoxOptions
- Reconnected IdsDropdownList into IdsDropdown and restored its previous behavior
- Provided examples/docs of how to use IdsDropdownList separately
- Integrated IdsDropdownList's usage into IdsDataGridFilters wherever dropdown is used.

**Related github/jira issue (required)**:
Closes #1065

**Steps necessary to review your pull request (required)**:

_Test dropdown integration:_
- open http://localhost:4300/ids-data-grid/filter-trigger-fields.html
- Use the columns that filter with Dropdowns.  Test that filtering works as expected
- Use the "Price" columns and filter on `13.99` (one option present in the grid).  Try opening the dropdown columns -- the dropdowns should not be cut off by Datagrid containment.

_Test separate list example:_
- open http://localhost:4300/ids-dropdown/separate-list.html
- use the Dropdown/DropdownList and make sure they behave like the others

_Smoke test previous behavior:_ check all these and make sure they work as expected:
- http://localhost:4300/ids-dropdown/example.html
- http://localhost:4300/ids-dropdown/sizes.html
- http://localhost:4300/ids-dropdown/typeahead.html
- http://localhost:4300/ids-dropdown/sandbox.html
- http://localhost:4300/ids-multiselect/example.html

**Included in this Pull Request**:
